From 77dd0437f2e4826c74a1e4155c8ae765d95490ea Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:45 -0400
Subject: [block] all blk-mq requests are tagged

Message-id: <1402670238-13916-118-git-send-email-snitzer@redhat.com>
Patchwork-id: 83839
O-Subject: [RHEL7.1 PATCH 117/150] block: all blk-mq requests are tagged
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit fb3ccb5da71273e7f0d50b50bc879e50cedd60e7
Author: Christoph Hellwig <hch@lst.de>
Date:   Mon Apr 14 10:30:12 2014 +0200

    block: all blk-mq requests are tagged

    Instead of setting the REQ_QUEUED flag on each of them just take it into
    account in the only macro checking it.

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 74d7ac9..48fdc9d 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -1157,7 +1157,8 @@ static inline bool blk_needs_flush_plug(struct task_struct *tsk)
 /*
  * tag stuff
  */
-#define blk_rq_tagged(rq)  ((rq)->cmd_flags & REQ_QUEUED)
+#define blk_rq_tagged(rq) \
+ ((rq)->mq_ctx || ((rq)->cmd_flags & REQ_QUEUED))
 extern int blk_queue_start_tag(struct request_queue *, struct request *);
 extern struct request *blk_queue_find_tag(struct request_queue *, int);
 extern void blk_queue_end_tag(struct request_queue *, struct request *);
-- 
1.7.1